pelican/generators.py

```diff
@@ -155,16 +155,21 @@ def get_files(self, paths, exclude=[], extensions=None):
 
             if os.path.isdir(root):
                 for dirpath, dirs, temp_files in os.walk(
-                        root, followlinks=True):
-                    drop = []
+                        root, topdown=True, followlinks=True):
                     excl = exclusions_by_dirpath.get(dirpath, ())
-                    for d in dirs:
+                    # We copy the `dirs` list as we will modify it in the loop:
+                    for d in list(dirs):
                         if (d in excl or
                             any(fnmatch.fnmatch(d, ignore)
                                 for ignore in ignores)):
-                            drop.append(d)
-                    for d in drop:
-                        dirs.remove(d)
+                            if d in dirs:
+                                dirs.remove(d)
+
+                    d = os.path.basename(dirpath)
+                    if (d in excl or
+                            any(fnmatch.fnmatch(d, ignore)
+                                for ignore in ignores)):
+                        continue
```

Comment:
Is this necessary? You won't get to an excluded folder because `dirs` above will filter any subfolders. And this will be in effect only if `root` of `os.walk` matches, which is odd...